omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -109,12 +113,32 @@ public Object create(SimpleObject post, RequestContext context) throws ResponseE
 				case UNLOAD:
 					unloadModules(modules, servletContext);
 					break;
+				case UPDATE:
+					updateModules(modules, action.getDownloadUrls());
+					break;
 			}
 		}
 		
 		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
 	}
 	
+	private void updateModules(Collection<Module> modules, Collection<String> downloadURLs) {
+		if (modules.isEmpty() || downloadURLs.isEmpty()) {
+			throw new IllegalRequestException("Modules or updateURLs can not be empty");
+		} else {
+			if (modules.size() == downloadURLs.size()) {
+				int index = 0;
+				ArrayList<String> downloadURLList = new ArrayList<String>(downloadURLs);
```

Comment:
Instead of constructing a new list, wouldn't you just look though the downloadURLs collection?

Reply:
Yah, That's a good idea. I will refactor that one